data_manager_select_index_by_path/data_manager/path_name_value_key_manager.py

```diff
@@ -71,22 +78,39 @@ def check_params(self):
         check_tab('index_path', str(self.index_path.absolute().name))
         check_tab('value', self.value)
         check_tab('dbkey', self.dbkey)
+        self.check_extra_columns()
+
+    def check_extra_columns(self):
+        index_properties = self.get_index_properties()
+        index_extra_columns = set(index_properties.get("extra_columns", []))
+        given_extra_columns = self.extra_columns.keys()
+        if index_extra_columns != given_extra_columns:
+            if len(index_extra_columns) > 0:
+                raise ValueError(
+                    "Values for the following columns should be "
+                    "supplied: {0}.".format(
+                        str(index_extra_columns).strip("{}")))
+            if len(index_extra_columns) == 0:
+                raise ValueError(
+                    "The table \'{0}\' does not have extra columns".format(
+                        self.data_table_name))
+        for key, value in self.extra_columns.items():
+            check_tab(key, value)
 
     def get_index_properties(self) -> dict:
         with self.indexes_properties_file.open('r') as properties_file:
             indexes = yaml.safe_load(properties_file)
         index_properties = indexes.get(self.data_table_name)
         if index_properties is None:
             raise ValueError(
-                "{0} not a supported table name".format(self.data_table_name))
+                "\'{0}\' not a supported table name".format(
+                    self.data_table_name))
         return index_properties
 
     def check_index_file_presence(self):
-        index_name = self.index_properties.get('name')
-        if index_name is None:
-            raise ValueError(
-                "Property 'name' not defined for '{0}',"
-                " please contact the developers to correct the mistake.")
+        index_name = self.index_properties.get(
+            'name',
+            '[Index name not found. Please report to developers]')
```

Comment:
I'm not sure storing the error message in this variable here is a good idea. If this property always has to be set, I would rather see it generates an exception with a sane error message (and it could potentially be caught during validation, before this block of code is ever hit).